crates/service/src/tap/receipt_store.rs

```diff
@@ -17,7 +18,25 @@ pub struct InnerContext {
 }
 
 impl InnerContext {
-    async fn store_receipts(&self, receipts: Vec<DatabaseReceipt>) -> Result<(), AdapterError> {
+    async fn process_db_receipts(&self, buffer: Vec<DatabaseReceipt>) {
+        let (v1_receipts, v2_receipts): (Vec<_>, Vec<_>) =
+            buffer.into_iter().partition_map(|r| match r {
+                DatabaseReceipt::V1(db_receipt_v1) => Either::Left(db_receipt_v1),
+                DatabaseReceipt::V2(db_receipt_v2) => Either::Right(db_receipt_v2),
+            });
+        let (insert_v1, insert_v2) = tokio::join!(
+            self.store_receipts_v1(v1_receipts),
+            self.store_receipts_v2(v2_receipts)
+        );
+        if let Err(e) = insert_v1 {
+            tracing::error!("Failed to store v1 receipts: {}", e);
+        }
+        if let Err(e) = insert_v2 {
+            tracing::error!("Failed to store v2 receipts: {}", e);
+        }
+    }
```

Comment:
```suggestion
    async fn process_db_receipts(&self, buffer: Vec<DatabaseReceipt>) {
        let (v1_receipts, v2_receipts): (Vec<_>, Vec<_>) =
            buffer.into_iter().partition_map(|r| match r {
                DatabaseReceipt::V1(db_receipt_v1) => Either::Left(db_receipt_v1),
                DatabaseReceipt::V2(db_receipt_v2) => Either::Right(db_receipt_v2),
            });
        let (insert_v1, insert_v2) = tokio::join!(
            self.store_receipts_v1(v1_receipts),
            self.store_receipts_v2(v2_receipts)
        );
        if let Err(e) = insert_v1 {
            tracing::error!("Failed to store v1 receipts: {}", e);
        }
        if let Err(e) = insert_v2 {
            tracing::error!("Failed to store v2 receipts: {}", e);
        }
    }
```

I like what you've done here, but let's return the error and have v1/v2 error variants, no?

Reply:
Should I return both errors? They are different things, one could process and the other could fail.

Reply:
Not sure about "should". I would imagine three enum/thiserror variants - with one of them being in the offchance of them both failing?

I guess the concern is that, as you say, all that could happen and the caller of the function could be absolutely none the wiser.

Reply:
tbh, I just moved this to a function because format and lsp are a mess within macros. The only caller for this function is inside the select!{} macro.

Reply:
done